raster-foundry/raster-foundry
# Add fields and filters to projects and annotations

## Overview

This PR adds below fields or filters and updates affected property tests to support annotate app frontend needs:
 - an `extras` field in `projects` table
 - `labeled_by` and `verified_by` fields in `annotations` table
 - a bounding box filter on annotations
 - tag filters: `tagsInclude` and `tagsExclude` to filter tagged projects in RF and in Annotate App

### Checklist

- ~Styleguide updated, if necessary~
- ~Any content changes are properly templated using `BUILDCONFIG.APP_NAME`~

### Notes

I will update spec in the morning.

## Testing Instructions

 * Run migrations
 * Check if `extras` field is added to `projects` table. Check CI. Do stuffs with projects to make sure no negative impacts.
 * Check if `labeled_by` and `verified_by` fields are added to `annotations` table. Check CI. Do stuffs with annotations to make sure no negative impacts.
 * Create some annotations. `GET /api/annotations?bbox=<a valid string of comma-delimited bbox in lat lng>` should get you annotations within this bounding box
 * Manually add some tags to some projects in db. 
   - `GET /api/projects?tagsExclude=<tag 1>&tagsExclude=<tag2>...` should get you projects without these tags
   - `GET /api/projects?tagsInclude=<tag 1>&tagsInclude=<tag2>...` should get you projects with these tags

Closes #3761
Closes #3762 
Closes #3763 
Closes #3764
